tests/common/mock_cgroup_commands.py

```diff
@@ -113,42 +129,37 @@ def mock_popen(command, *args, **kwargs):
         return original_popen(command, *args, **kwargs)
 
     def get_mapped_path(path):
-        for item in patcher.files:
+        for item in __MOCKED_FILES:
             match = re.match(item[0], path)
             if match is not None:
                 return item[1]
-        for item in __SYSTEM_PATHS:
-            mapped = re.sub(r"^({0})".format(item), r"{0}\1".format(tmp_dir), path)
+
+        for item in __MOCKED_PATHS:
+            mapped = re.sub(item, r"{0}\1".format(tmp_dir), path)
             if mapped != path:
                 mapped_parent = os.path.split(mapped)[0]
                 if not original_path_exists(mapped_parent):
                     os.makedirs(mapped_parent)
                 return mapped
         return path
 
-    def mock_read_file(filepath, **kwargs):
-        return original_read_file(get_mapped_path(filepath), **kwargs)
-
-    def mock_write_file(filepath, content, **kwargs):
-        return original_write_file(get_mapped_path(filepath), content, **kwargs)
-
     def mock_mkdir(path, *args, **kwargs):
         return original_mkdir(get_mapped_path(path), *args, **kwargs)
 
+    def mock_open(path, *args, **kwargs):
+        return original_open(get_mapped_path(path), *args, **kwargs)
+
     def mock_path_exists(path):
         return original_path_exists(get_mapped_path(path))
 
+    builtin_popen = "__builtin__.open" if sys.version_info[0] == 2 else "builtins.open"
```

Comment:
This seems like a good candidate for future.py for a global use. What do you think?

Reply:
Probably test tools, since it is meant for test code. I can create a patch_builtin function there.

Different PR, though :)